python/pylibcugraph/pylibcugraph/uniform_neighborhood_sampling.pyx

```diff
@@ -0,0 +1,190 @@
+    bool_t,
+    data_type_id_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_free,
+    cugraph_type_erased_host_array_view_t,
+    cugraph_type_erased_host_array_view_create
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph._cugraph_c.algorithms cimport (
+    uniform_nbr_sample,
+    cugraph_sample_result_t,
+    cugraph_sample_result_get_sources,
+    cugraph_sample_result_get_destinations,
+    cugraph_sample_result_get_start_labels,
+    cugraph_sample_result_get_index,
+    cugraph_sample_result_get_counts,
+    cugraph_sample_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    EXPERIMENTAL__ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    copy_to_cupy_array,
+    assert_CAI_type,
+    get_c_type_from_numpy_type,
+)
+
+def EXPERIMENTAL__uniform_neighborhood_sampling(EXPERIMENTAL__ResourceHandle resource_handle,
+                               _GPUGraph input_graph,
+                               start_info_list,
+                               h_fan_out,
+                               bool_t with_replacement,
+                               bool_t do_expensive_check):
+    """
+    Does uniform neighborhood sampling.
+
+    Parameters
+    ----------
+    resource_handle: ResourceHandle
+        Handle to the underlying device and host resources needed for
+        referencing data and running algorithms.
+
+    input_graph: MGGraph
+        The input graph, for Multi-GPU operations.
+
+    start_info_list: device array type
+        Device array containing the list of starting vertices for sampling,
+        as well as starting labels for reorganizing the results after the
+        sending the input to different callers
+
+    fanout_vals: list
+        Host array containing the fan out at each step in the sampling
+        algorithm
+
+    with_replacement: bool
+        If true, sampling procedure is done with replacement (the same vertex
+        can be selected multiple times in the same step)
+
+    do_expensive_check: bool
+        If True, performs more extensive tests on the inputs to ensure
+        validitity, at the expense of increased run time.
+
+    Returns
+    -------
+    A tuple of device arrays, where the first and second items in the tuple
+    are device arrays containing the starting and ending vertices of each
+    walk respectively, the third item in the tuple is a device array
+    containing the start labels, the fourth item in the tuple is a device
+    array containing the indices for ..., and the fifth item in the tuple
+    is a device containing the counts for ...
+
+    Examples
```

Comment:
This example won't work because you need to distribute the edges across workers(GPUs) and for each partition, call `pylibcugraph.experimental.MGGraph` . We do this by leveraging `dask`. I think you should remove the example for now until we find a solution for that.

Reply:
Yeah, I agree about the part where you said to remove an example, the cugraph wrapper too wouldn't benefit from an example if it can't be tested using doctest